### PR TITLE
docs: update glossary to include the term template-input-variable

### LINKED ISSUE #39783: Add template input variable to glossary
# 📚 Docs or angular.io bug report

### Description

Currently, we do not have glossary entry for template input variable.

### What's the affected URL?**
Anywhere where template input variables are referenced in the docs.

## 🔥 Exception or Error
<pre><code>

</code></pre>

### Anything else relevant?

It would be very helpful if we could link to a glossary definition whenever we mention template input variables.

### PR DESCRIPTION
**Pull request by andrewasche**
Fixes #39783

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/glossary.md [hunk 1]
```diff
@@ -964,6 +964,14 @@ The following example defines a template reference variable named `#phone`.
 
 For more information, see the [Template reference variable](guide/template-reference-variables) guide.
 
+{@a template-input-variable}
+
+## template input variable
+
+A template input variable is a variable you can reference within a single instance of the template. You declare a template input variable using the let keyword as in let hero.
+
+Read and learn more about [Template Input Variables](template-reference-variables#template-input-variable).
```
Comment (AndrewKushnir) [inline aio/content/guide/glossary.md:973]: It looks like `guide/` is missing:
```suggestion
Read and learn more about [Template Input Variables](guide/template-reference-variables#template-input-variable).
```

File: aio/content/guide/glossary.md [hunk 2]
```diff
@@ -964,6 +964,14 @@ The following example defines a template reference variable named `#phone`.
 
 For more information, see the [Template reference variable](guide/template-reference-variables) guide.
 
+{@a template-input-variable}
+
+## template input variable
+
+A template input variable is a variable you can reference within a single instance of the template. You declare a template input variable using the let keyword as in let hero.
```
Comment (AndrewKushnir) [inline aio/content/guide/glossary.md:971]:
```suggestion
A template input variable is a variable you can reference within a single instance of the template. You declare a template input variable using the `let` keyword as in `let hero`.
```

**Review by AndrewKushnir — COMMENTED** [review]
@andrewasche thanks for creating this PR, I've left a couple comments, please see below.

I think it'd also be beneficial to have a quick example of the template input variable (similar to an example available in the "template reference variable" section of the glossary) and may be also use the same example in the guide that you refer to: https://angular.io/guide/template-reference-variables#template-input-variable

Without an example it's not obvious what the "single instance of the template" refers to.

File: aio/content/guide/glossary.md (hunk 2 above)
Comment (petebacondarwin) [reply aio/content/guide/glossary.md:971]: Why was this resolved without making the change nor a comment to say why not?

File: aio/content/guide/glossary.md (hunk 1 above)
Comment (petebacondarwin) [reply aio/content/guide/glossary.md:973]: Why was this resolved without making the change nor a comment to say why not?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Please can you address @AndrewKushnir's comments?

**Comment by andrewasche** [comment]
Sorry about my confusion. I added a code example, I missed the getting the `let` correct. I'm working on getting that too, but ran into an issue.

**Comment by andrewasche** [comment]
okay, I think that last one is good. think the changes for `let` and the hyperlink are good. and I added a example which I think is appropriate

**Comment by petebacondarwin** [comment]
@andrewasche - Did you forget to push your commits. There doesn't appear to be any changes since the reviews.

**Comment by andrewasche** [comment]
I ran into issues again, I think I didn't sync the files properly, but i did get another commit in

File: aio/content/guide/component-styles.md [hunk 3]
```diff
@@ -235,7 +236,7 @@ When building with the CLI, you must configure the `angular.json` to include _al
 
 Register **global** style files in the `styles` section which, by default, is pre-configured with the global `styles.css` file.
 
-See the [CLI wiki](https://github.com/angular/angular-cli/wiki/stories-global-styles) to learn more.
+See the [Styles configuration guide](https://angular.io/guide/workspace-config#styles-and-scripts-configuration) to learn more.
```
Comment (petebacondarwin) [inline aio/content/guide/component-styles.md:239]:
```suggestion
See the [Styles configuration guide](https://angular.io/guide/workspace-config#styles-and-scripts-configuration) to learn more.
```
```suggestion
See the [Styles configuration guide](guide/workspace-config#styles-and-scripts-configuration) to learn more.
```

File: aio/content/guide/component-styles.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 Angular applications are styled with standard CSS. That means you can apply
 everything you know about CSS stylesheets, selectors, rules, and media queries
+
```
Comment (petebacondarwin) [inline aio/content/guide/component-styles.md:5]: ?

File: aio/content/guide/glossary.md [hunk 5]
```diff
@@ -912,6 +912,28 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+{@a template-input-variable}
+
+## template input variable
+
+A template input variable is a variable you can reference within a single instance of the template. You declare 
+a template input variable using the `let` keyword as in `let hero`.
+
+```
+<ng-template [ngIf]="true">
+  <!-- The reference is defined within a template -->
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:925]:
```suggestion
  <!-- The `ref2` reference is only defined within this template -->
```

File: aio/content/guide/glossary.md [hunk 6]
```diff
@@ -912,6 +912,28 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+  <input #ref2 type="text" [(ngModel)]="secondExample" />
+</ng-template>
+<!-- ref2 accessed from outside that template doesn't work -->
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:928]:
```suggestion
<!-- `ref2` accessed from outside its template doesn't work -->
```

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks for the updates @andrewasche  - nearly there!

**Comment by andrewasche** [comment]
Thanks for your help! I feel it coming together.

File: aio/content/guide/component-styles.md [hunk 7]
```diff
@@ -235,7 +236,7 @@ When building with the CLI, you must configure the `angular.json` to include _al
 
 Register **global** style files in the `styles` section which, by default, is pre-configured with the global `styles.css` file.
 
-See the [CLI wiki](https://github.com/angular/angular-cli/wiki/stories-global-styles) to learn more.
+See the [Styles configuration guide](guide/workspace-config#styles-and-scripts-configuration) to learn more.
```
Comment (petebacondarwin) [inline aio/content/guide/component-styles.md:239]: Should we remove the comment from below this line?

File: aio/content/guide/component-styles.md (hunk 4 above)
Comment (petebacondarwin) [reply aio/content/guide/component-styles.md:5]: Can you remove this change.

File: aio/content/guide/glossary.md [hunk 8]
```diff
@@ -912,6 +912,28 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+a template input variable using the `let` keyword as in `let hero`.
+
+```
+<ng-template [ngIf]="true">
+  <!-- The `ref2` reference is only defined within this template -->
+  <input #ref2 type="text" [(ngModel)]="secondExample" />
+</ng-template>
+<!-- `ref2` accessed from outside its template doesn't work -->
+<span>Value: {{ ref2?.value }}</span>
+```
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:930]: I just realised that this example doesn't match the text above it.
This example is showing a template reference variable, not a template input variable.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

File: aio/content/guide/component-styles.md (hunk 7 above)
Comment (andrewasche) [reply aio/content/guide/component-styles.md:238]: looks like that  comment was for the last link, so I deleted it

Comment (andrewasche) [reply aio/content/guide/component-styles.md:238]: I think it is for the previous link I would assume it is not needed.

**Review by andrewasche — COMMENTED** [review]
I think I have address some issues. I got rid of extraneous space and switch the example to use template input variable, customer.

File: aio/content/guide/component-styles.md (hunk 7 above)
Comment (petebacondarwin) [reply aio/content/guide/component-styles.md:238]: Please go ahead and remove the comment.

File: aio/content/guide/glossary.md [hunk 9]
```diff
@@ -912,6 +912,37 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+{@a template-input-variable}
+
+##  <tr *ngFor="let customer of customers;">
+     <td>{{customer.customerNo}}</td>
+     <td>{{customer.name}}</td>
+     <td>{{customer.address}}</td>
+     <td>{{customer.city}}</td>
+     <td>{{customer.state}}</td>
+     <button (click)="selectedCustomer=customer">Select</button>
+   </tr> variable
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:925]: This doesn't look right 😸

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

File: aio/content/guide/glossary.md (hunk 9 above)
Comment (andrewasche) [reply aio/content/guide/glossary.md]: I realized my grave mistake, and change it to match the format of the glossary

Comment (andrewasche) [reply aio/content/guide/glossary.md]: I also got rid of the comment from 2018

File: aio/content/guide/glossary.md [hunk 10]
```diff
@@ -912,6 +912,30 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
 
 Read about how to write template expressions in  [Template expressions](guide/template-syntax#template-expressions).
 
+
+{@a template-input-variable}
+
+## template input variable
+
+A template input variable is a variable you can reference within a single instance of the template. You declare 
+a template input variable using the `let` keyword as in `let hero`.
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:921]: Given that the following example snippet does not have `hero`, then I suggest:

```suggestion
a template input variable using the `let` keyword as in `let customer`.
```

File: aio/content/guide/glossary.md [hunk 11]
```diff
@@ -912,6 +912,30 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+     <td>{{customer.name}}</td>
+     <td>{{customer.address}}</td>
+     <td>{{customer.city}}</td>
+     <td>{{customer.state}}</td>
+     <button (click)="selectedCustomer=customer">Select</button>
+   </tr>
+```
+
+Read and learn more about [template input variables](guide/template-reference-variables#template-input-variable)
+.
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:935]:
```suggestion
Read and learn more about [template input variables](guide/template-reference-variables#template-input-variable).
```

File: aio/content/guide/glossary.md [hunk 12]
```diff
@@ -912,6 +912,30 @@ A TypeScript-like syntax that Angular evaluates within a [data binding](#data-bi
...
+
+
+
```
Comment (petebacondarwin) [inline aio/content/guide/glossary.md:938]: One too many blank lines?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

**Comment by petebacondarwin** [comment]
The component-styles.md had already been fixed three weeks ago.
So I have rebased and squashed all the other commits.

**Comment by petebacondarwin** [comment]
@googlebot I consent.

**Comment by petebacondarwin** [comment]
@googlebot I consent.

**Review by petebacondarwin — APPROVED** [review]

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: global-docs-approvers, public-api

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: public-api

**Pull request closed without merging**